lib/heex_formatter/phases/format.ex

```diff
@@ -77,23 +82,42 @@ defmodule HeexFormatter.Phases.Format do
         end
       end
 
-    indentation = if self_closed?, do: state.indentation, else: state.indentation + 1
+    state =
+      if self_closed? do
+        state
+      else
+        if tag == "script" do
+          %{state | mode: :script}
+        else
+          %{state | indentation: state.indentation + 1}
+        end
+      end
```

Comment:
Would the code below cover it too? Maybe `cond` would be more clear here
```suggestion
cond do
  self_closed? and tag == "script" ->
    %{state | mode: :script}
    
  self_closed? ->
    state
    
  true ->
    %{state | indentation: state.indentation + 1}
end
```

Reply:
I guess we can avoid this `mode` and pattern match in the previous token. Not 100% yet, I'm trying something out here 🙌🏻